core/primitives/src/types.rs

```diff
@@ -152,6 +154,28 @@ pub enum StateChangeCause {
     /// State change that happens when we update validator accounts. Not associated with with any
     /// specific transaction or receipt.
     ValidatorAccountsUpdate,
+    /// State change that is happens due to migration that happens in first block of an epoch
+    /// after protocol upgrade
+    Migration { migration_id: MigrationId },
+}
+
+#[derive(Serialize, Deserialize, Clone, BorshSerialize, BorshDeserialize)]
+pub enum MigrationId {
+    StorageUsageFix,
```

Comment:
Can we add a comment on what the fix is (maybe provide some link as well)?

Reply:
Done